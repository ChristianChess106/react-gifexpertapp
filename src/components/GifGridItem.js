import React from 'react'

export const GifGridItem = ({id,title,url}) => {

    /* console.log(id,title); */
    return (
        <div className="card animate__fadeInDown">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}
