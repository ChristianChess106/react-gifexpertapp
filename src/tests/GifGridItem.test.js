import { shallow } from "enzyme"
import { GifGridItem } from "../components/GifGridItem"

describe('Mostrar el componente Correctamente GifGridItem', () => {
    
    const titulo = "un titulo";
    const url = "https://localhost/algo.jpg";

    let wrapper = shallow(<GifGridItem title={titulo} url={url}/>);
   
    
    test('Componente GifGridItem ', () => {

      //  console.log(wrapper.props());
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de tener un parrafo con el titulo ', () => {

        const p = wrapper.find("p");
        expect(p.text().trim()).toBe(titulo);
    });

    test('debe tener imagen igual al url y alt de los props ', () => {
        
        const img = wrapper.find("img");
        //console.log(img.props("src"));

        expect(img.prop("src")).toBe(url);
        expect(img.prop("alt")).toBe(titulo);
    });
    test('debe tener card animate__fadeInUpBig ', () => {
        
        const div = wrapper.find("div");

        expect(div.prop("className")).toBe("card animate__fadeInUpBig");
        
    });
    
    
    
    
})
