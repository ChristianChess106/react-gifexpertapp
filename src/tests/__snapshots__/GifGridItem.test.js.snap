// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mostrar el componente Correctamente GifGridItem Componente GifGridItem  1`] = `
<div
  className="card animate__fadeInUpBig"
>
  <img
    alt="un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    un titulo
  </p>
</div>
`;
